### PR TITLE
Move audio backends into seperate crate

### PR DESCRIPTION
**Pull request by sashahilton00**
This just moves the audio backends into a seperate crate, and corrects the feature flag from ```dns-sd``` to ```with-dns-sd```.

### CODE REVIEW

**Comment by sashahilton00** [comment]
It seems cross compilation for ARM has been broken. Just tried to run a cross compile with docker and got the same outcome, backtrace included below. Would appreciate another set of eyes on this, as this change was just reorganising files, hence it's unclear to me what's causing the cross compilation failure on Travis all of a sudden.

```
docker run --env "PKG_CONFIG_ALLOW_CROSS=1" --env "RUST_BACKTRACE=full" -v /tmp/librespot-build:/build librespot-cross cargo build --target arm-unknown-linux-gnueabihf --no-default-features
   Compiling portaudio-sys v0.1.1
   Compiling syn v0.11.11
   Compiling lewton v0.8.0
   Compiling syntex_errors v0.58.1
error: failed to run custom build command for `portaudio-sys v0.1.1`
process didn't exit successfully: `/build/debug/build/portaudio-sys-223ffc1e679b014f/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Failed to run `"pkg-config" "--libs" "--cflags" "portaudio-2.0"`: No such file or directory (os error 2)', /build/cache/registry/src/github.com-1ecc6299db9ec823/portaudio-sys-0.1.1/build.rs:8:18
stack backtrace:
   0:     0x5630523997c3 - std::sys::imp::backtrace::tracing::imp::unwind_backtrace::hd278582b7bf125b4
                               at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1:     0x563052395dcc - std::sys_common::backtrace::_print::h585aeaf99da81bd9
                               at /checkout/src/libstd/sys_common/backtrace.rs:68
   2:     0x56305239dc21 - std::panicking::default_hook::{{closure}}::h2cf5ae2c765258ee
                               at /checkout/src/libstd/sys_common/backtrace.rs:57
                               at /checkout/src/libstd/panicking.rs:381
   3:     0x56305239d997 - std::panicking::default_hook::h4df9c6962fc8f370
                               at /checkout/src/libstd/panicking.rs:397
   4:     0x56305239e06a - std::panicking::rust_panic_with_hook::hec57cb9076b0053e
                               at /checkout/src/libstd/panicking.rs:577
   5:     0x56305239deee - std::panicking::begin_panic::hf1e5e0b8a9c20d4e
                               at /checkout/src/libstd/panicking.rs:538
   6:     0x56305239dde9 - std::panicking::begin_panic_fmt::hae9efc5efbaf8262
                               at /checkout/src/libstd/panicking.rs:522
   7:     0x5630523726f7 - build_script_build::main::hfef0d819b0f4b844
                               at /build/cache/registry/src/github.com-1ecc6299db9ec823/portaudio-sys-0.1.1/build.rs:8
   8:     0x5630523a65fe - __rust_maybe_catch_panic
                               at /checkout/src/libpanic_unwind/lib.rs:101
   9:     0x56305239e60b - std::rt::lang_start::hee06cb52bfc299e1
                               at /checkout/src/libstd/panicking.rs:459
                               at /checkout/src/libstd/panic.rs:365
                               at /checkout/src/libstd/rt.rs:58
  10:     0x56305237274d - main
  11:     0x7f0076c7e2b0 - __libc_start_main
  12:     0x563052372229 - _start
  13:                0x0 - <unknown>

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by sashahilton00** [comment]
Now 3am, will correct this tomorrow.

**Comment by sashahilton00** [comment]
Ok, can anyone else see why cross compilation is failing in this PR? I've basically just moved files into a new crate, so struggling to see why it isn't cross-compiling, yet it's compiling fine when run natively...

File: playback/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jack            = { version = "0.5.3", optional = true }
+libc            = { version = "0.2", optional = true }
+
+[features]
+alsa-backend = ["alsa"]
+portaudio-backend = ["portaudio-rs"]
+pulseaudio-backend = ["libpulse-sys", "libc"]
+jackaudio-backend = ["jack"]
+
+default = ["portaudio-backend"]
```
Comment (plietar) [inline playback/Cargo.toml]: This is causing the cross-compilation to fail.

We don't build any backends on Travis when cross-compiling, but `--no-default-features` does not propagate to dependencies.

Having `default = ["librespot-playback/portaudio-backend"]` in the root `Cargo.toml` is enough.

Comment (sashahilton00) [reply playback/Cargo.toml]: Good to know, thanks for pointing that out. Will let Travis run and merge if successful.

**Pull request merged**